subsys/bluetooth/host/settings.c

```diff
@@ -119,6 +119,14 @@ static int set(const char *name, size_t len_rd, settings_read_cb read_cb,
 	ssize_t len;
 	const char *next;
 
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_ENABLE)) {
+		/* Cannot load settings before bt_enable() because this handler uses bt
+		 * functions. The application shall call settings_load_subtree("bt")
+		 * after bt_enable().
+		 */
+		return 0;
```

Comment:
I don't think returning 0 in case that setting were not loaded is very good. Perhaps `-ECANCELED`?

Reply:
As this is no error situation but intended behavior it should not return an error.

Reply:
If we want to stick with the current approach I would at least add an warning log to ensure that user is actually aware that his Bluetooth settings were not loaded even though he called globally settings_load.

Reply:
Adding a warning or debug message inside the `bt` settingshandler routines is fine.

Delegating this to the `settings` system is bad: log messages wil only be seen if the logsystem is enabled for `settings` (and not `bt`), the `settings` system cannot return this to the user because it might have received different error messages from different subsystems.